File: packages/py/tracing/examples/langgraph.py
from typing_extensions import TypedDict
from langgraph.graph import END, START, StateGraph
from uselemma_tracing import langgraph


class GraphState(TypedDict):
    input: str
    output: str


def answer(state: GraphState):
    return {"output": f"You said: {state['input']}"}


graph = (
    StateGraph(GraphState)
    .add_node("answer", answer)
    .add_edge(START, "answer")
    .add_edge("answer", END)
    .compile()
)


def call_langgraph(user_message: str):
    result = graph.invoke(
        {"input": user_message},
        {"callbacks": [langgraph(agent_name="support-graph")]},
    )
    return result["output"]

File: packages/ts/tracing/examples/langgraph.ts
import { StateGraph, START, END } from "@langchain/langgraph";
import { langGraph } from "@uselemma/tracing";

type GraphState = {
  input: string;
  output?: string;
};

const graph = new StateGraph<GraphState>()
  .addNode("answer", async (state) => ({
    output: `You said: ${state.input}`,
  }))
  .addEdge(START, "answer")
  .addEdge("answer", END)
  .compile();

export async function callLangGraph(userMessage: string) {
  const result = await graph.invoke(
    { input: userMessage },
    { callbacks: [langGraph({ agentName: "support-graph" })] },
  );

  return result.output;
}
